api/src/main/java/org/openmrs/module/fhir/api/diagnosticreport/handler/LaboratoryHandler.java

```diff
@@ -150,12 +153,25 @@ public DiagnosticReport saveFHIRDiagnosticReport(DiagnosticReport diagnosticRepo
 	}
 
 	@Override
-	public DiagnosticReport purgeFHIRDiagnosticReport(DiagnosticReport diagnosticReport) {
-		// Delete `Name` Obs
-		// Delete `Status` Obs
-
-		// Delete Obs (`Result` as Set of Obs)
+	public void retireFHIRDiagnosticReport(String id) {
+		EncounterService encounterService = Context.getEncounterService();
 		// Delete Binary Obs Handler which used to store `PresentedForm`
-		return diagnosticReport;
+
+		// Delete Encounter OpenMRS Object
+		Encounter omrsDiagnosticReport = encounterService.getEncounterByUuid(id);
+
+		if (omrsDiagnosticReport == null) {
```

Comment:
@harsha89 , as per the API, I do believe that the underlying Obs are retired when you retire an encounter. At least, that's what happens according to: http://resources.openmrs.org/doc/org/openmrs/api/EncounterService.html#voidEncounter-org.openmrs.Encounter-java.lang.String-